### PR TITLE
fix: min/max width/height enforcement when `aspectRatio` is set

### PR DESCRIPTION
**Pull request by codebytere**
#### Description of Change

Closes https://github.com/electron/electron/issues/37305.

Fixes an issue where `minWidth`/`minHeight` and `maxWidth`/`maxHeight` would not be enforced if the user set an `aspectRatio` on macOS. This was happening because we were explicitly setting the  new width and height in `windowWillResize`, which overrides the handling macOS sets on the updated frame when `setContentMinSize` and `setContentMaxSize` were previously called on the `NSWindow`.

cc @carterbs 

#### Checklist

- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] [PR release notes](https://github.com/electron/clerk/blob/main/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/main/README.md#examples).

#### Release Notes

Notes: Fixes an issue where `minWidth`/`minHeight` and `maxWidth`/`maxHeight` would not be enforced if the user set an `aspectRatio` on macOS.

### CODE REVIEW

File: shell/browser/ui/cocoa/electron_ns_window_delegate.mm [hunk 1]
```diff
@@ -128,27 +128,62 @@ - (void)windowDidResignKey:(NSNotification*)notification {
...
+        shell_->has_frame() ? NSMakeSize(0, titleBarHeight) : NSZeroSize;
+    if (!NSEqualSizes(minSize, zeroSize)) {
+      double minWidthForAspectRatio =
+          (minSize.height - titleBarHeight) * aspectRatio;
+      bool atMinHeight =
+          minSize.height > zeroSize.height && newSize.height <= minSize.height;
+      newSize.width = atMinHeight ? minWidthForAspectRatio
+                                  : std::max(newSize.width, minSize.width);
+
+      double minHeightForAspectRatio = minSize.width / aspectRatio;
```
Comment (carterbs) [inline shell/browser/ui/cocoa/electron_ns_window_delegate.mm:162]: I don't know how verbose comments in electron are, but I had to read this and actually derive these values to believe them. Might be helpful to have the formula:
 
If aspect ratio is r and `r = w/h`
Then `w = r*h`
and `h = w/r`

Comment (codebytere) [reply shell/browser/ui/cocoa/electron_ns_window_delegate.mm:162]: > actually derive these values to believe them

Can you explain what you mean here? What kind of comment would you expect to see?

Comment (carterbs) [reply shell/browser/ui/cocoa/electron_ns_window_delegate.mm:162]: I'm an excessive commenter so take this with a grain of salt - I know all devs and all projects have different norms and commenting debates can get spicy.

When I initially read the code I was surprised that we were using width to derive the height of the window. Of course, that's necessary when locking the aspect ratio. But I haven't calculated an aspect ratio in a while, so I looked up the formula and went from there. After that, it made sense.

If I were writing this code, I'd just note the formula for other people who forgot that `r = w/h`. But I think it's equally valid to put no comments.

**Review by zcbenz — APPROVED** [review]

**Review by jkleinsc — COMMENTED** [review]
Change looks good to me, but can we add test(s)?

**Review by jkleinsc — APPROVED** [review]

**Pull request merged**